Godeps/_workspace/src/github.com/samuel/go-zookeeper/zk/conn.go

```diff
@@ -454,9 +454,6 @@ func (c *Conn) authenticate() error {
 		return ErrSessionExpired
 	}
 
-	if c.sessionID != r.SessionID {
-		atomic.StoreUint32(&c.xid, 0)
-	}
```

Comment:
Can you re-sync this to master rather than leaving a diff here? It might avoid some surprises when updating in the future.

```
$ go get -u github.com/samuel/go-zookeeper/zk
$ godep update github.com/samuel/go-zookeeper/zk
```